actionmailbox/app/controllers/action_mailbox/ingresses/mailgun/inbound_emails_controller.rb

```diff
@@ -64,14 +64,22 @@ def authenticated?
           ).authenticated?
         else
           raise ArgumentError, <<~MESSAGE.squish
-            Missing required Mailgun API key. Set action_mailbox.mailgun_api_key in your application's
-            encrypted credentials or provide the MAILGUN_INGRESS_API_KEY environment variable.
+            Missing required Mailgun Signing key. Set action_mailbox.mailgun_signing_key in your application's
+            encrypted credentials or provide the MAILGUN_INGRESS_SIGNING_KEY environment variable.
           MESSAGE
         end
       end
 
       def key
-        Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key) || ENV["MAILGUN_INGRESS_API_KEY"]
+        if Rails.application.credentials.dig(:action_mailbox, :mailgun_api_key) ||  ENV["MAILGUN_INGRESS_API_KEY"]
+          ActiveSupport::Deprecation.warn(<<-MSG.squish)
+            Rails.application.credentials.action_mailbox.api_key and ENV.MAILGUN_INGRESS_API_KEY are deprecated and will be removed from Rails x.x.
```

Comment:
This deprecation will be added to 6.1, so the message should mention 6.2:

```suggestion
            Rails.application.credentials.action_mailbox.api_key and ENV.MAILGUN_INGRESS_API_KEY are deprecated and will be removed from Rails 6.2.
```